### PR TITLE
nit(attribute-validation): Using switch instead of multiple if checks.

### PR DESCRIPTION
**Pull request by yasirfolio3**
## Fixes:
- Using switch instead of multiple if checks.
- Covered all go basic types defined here: https://go.dev/tour/basics/11

## Tests:
- All tests should pass

## Jira ticket:
https://optimizely.atlassian.net/browse/OASIS-8593

### CODE REVIEW

File: pkg/event/factory.go [hunk 1]
```diff
@@ -297,17 +297,10 @@ func isValidAttribute(value interface{}) bool {
...
-	}
-	if _, ok := value.(float64); ok {
-		return true
-	}
-	if _, ok := value.(int); ok {
-		return true
-	}
-	if _, ok := value.(bool); ok {
+	switch value.(type) {
+	case string, float64, int, bool:
```
Comment (jaeopt) [inline pkg/event/factory.go:301]: Does this change work for float32, int8, which should be accepted?

**Review by jaeopt — CHANGES_REQUESTED** [review]
A  clarification

File: pkg/event/factory.go (hunk 1 above)
Comment (msohailhussain) [reply pkg/event/factory.go:301]: https://go.dev/tour/basics/11
should give all types?

Comment (yasirfolio3) [reply pkg/event/factory.go]: have covered all basic types now, please check 👍

**Review by jaeopt — APPROVED** [review]
LGTM
wondering what's wrong with the original if-based one. It'll cover all types.

**Comment by yasirfolio3** [comment]
> LGTM wondering what's wrong with the original if-based one. It'll cover all types.

i tried the new tests for both new and previous code, behaviour was the same. We would've required `if` checks for all types too.

**Review by jaeopt — APPROVED** [review]
LGTM

**Review by msohailhussain — APPROVED** [review]
lgtm

**Pull request merged**